cosign/README.md

```diff
@@ -0,0 +1,125 @@
+3. Publish a long-lived signature over an artifact (or artifact meta-data) to an Artifact Log, allowing verifiers to check its validity.
+
+The first and second operations are performed by a system called Fulcio, which functions as a Certificate Authority and transparency log for a namespace of OIDC identities (Identity Log).
+The third operation is performed by a system called Rekor, a transparency log for artifact signatures (Artifact Log).
+And cosign is the reference Sigstore client implementation to sign arbitrary artifacts. With these components, Sigstore completed trust setup, signing and verification[2].
+
+```mermaid
+flowchart TB
+  %% OIDC Flow
+  Signer["💻 Signer"] -->|① OIDC Challenge Start| OIDC["@ OIDC Provider"]
+  OIDC -->|② OIDC Response| Signer
+  
+  %% Certificate Issuance via Fulcio
+  Signer -->|③ Issuance Request| Fulcio["📜 Fulcio (Certificate Authority)"]
+  Fulcio -->|④ Signed Certificate| Signer
+  Fulcio -->|Publish to Identity Log| IdentityLog
+  
+  %% Signing & Publishing Artifacts
+  Signer -->|⑤ Sign and Publish Artifact| ArtifactLog
+  Signer -->|⑤ Sign and Publish Artifact| ArtifactRepo["📦 Artifact Repository"]
+  
+  %% Identity and Artifact Logs
+  subgraph Logs["Transparency Logs"]
+    subgraph IdentityLog["📜 Identity Log (Fulcio)"]
+      ID1["UUID: 5a88c7 SAN: user#64;github.com"]
+      ID2["UUID: 7a40cf SAN: user2#64;github.com"]
+      ID3["UUID: b76141 SAN: user#64;name.io"]
+    end
+    
+    subgraph ArtifactLog["📝 Artifact Log (Rekor)"]
+      AID1["UUID: 5a88c7 SAN: io.github.pytorch"]
+      AID2["UUID: 7a40cf SAN: com.aws.xxx"]
+      AID3["UUID: b76141 SAN: dev.client.artifact"]
+    end
+  end
+
+  %% Artifact Verification Flow
+  ArtifactRepo -->|⑧ Download the Artifact| Verifier["🔍 Verifier"]
+  ArtifactLog -->|⑦ Fetch Artifact Signature|Verifier
+  IdentityLog -->|⑥ Update Root of Trust|Verifier  
+
+```
+
+To verify an artifact, a verifier ensures that the trusted keys are fresh by performing a key update. With an updated root-of-trust, the first check ensures that the identity claimed was correctly verified by the identity log, and that the signature provided matches the content of the signed artifact. A final check verifies that the signature was created during the time window in which the certificate was valid.
+
+The Sigstore team plans to refactor parts of cosign into a new, minimal, and user-friendly API named `sigstore-go`. Currently in beta, `sigstore-go` has passed the `sigstore-conformance` signing and verification test suite.
+`sigstore-go` is built as a modular Go codebase including cryptographic signature creation and validation, integration with external services (Fulcio for certificates and Rekor for record-keeping) and API layers.
+The `ratify-verifier-go` implementation shares the same underlying library, `sigstore-go`, as the cosign verifier.
+
+## Concepts
```

Comment:
Timestamping is not mentioned, which is a key concept in the cosign keyless signing / verification scenario.

Reply:
Sure.